src/openassetio-core/tests/deprecationsTest.cpp

```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2023 The Foundry Visionmongers Ltd
+#include <type_traits>
+
+#include <catch2/catch.hpp>
+
+#include <openassetio/TraitsData.hpp>
+#include <openassetio/trait/TraitsData.hpp>
+
+// Tests for deprecated functionality.
```

Comment:
[fix] Fails due to `-Werror` on gcc and clang (including clang-tidy - and it's not a check, it's a compiler error, so no `.clang-tidy` config fix). The following appears to work. Not tested on MSVC, but we _still_ haven't configured compiler warnings for MSVC (perhaps one for the CI sprint?)

```suggestion
// Disable deprecation warning that triggers -Werror, failing the build.
#if defined(__clang__)
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
#elif defined(__GNUC__)
#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
#endif

// Tests for deprecated functionality.
```